### PR TITLE
Refactoring of the converter for Activation operators.

### PR DESCRIPTION
**Pull request by drivanov**
This PR Refactors the converter for Activation operators. It also replaces the special implemented converters for `Relu6` and `LeakyRelu`  operator with the activation operator implementation.

Add a new check and corresponding subtests for Validation: at least 1 dimension is required for input of any Activation operation.

### CODE REVIEW

**Comment by gbaned** [comment]
@drivanov Can you please resolve conflicts? Thanks!

**Comment by bixia1** [comment]
I edited the PR description, please check.

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 1]
```diff
@@ -1370,20 +1371,34 @@ class OpConverterTest : public ::testing::Test {
     }
   }
 
-  // Add weights for both validation and conversion.
-  template <typename T>
+  // Adds weights for both validation and conversion. The type of the weight is
+  // determined by tf_type. The initial value vector (values) can have any
+  // type (T) that can be static casted to tf_type.
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:1376]: s/static/statically/

File: tensorflow/compiler/tf2tensorrt/convert/ops/unary_ops.cc [hunk 2]
```diff
@@ -54,53 +54,191 @@ UnaryOperationMap() {
...
+            "Unary op: '", op, "' is not supported in implicit batch mode");
+      }
     }
 
     return Status::OK();
   }
 
-  Status Convert() {
-    const auto& params = *this->params_;
+  Status ImplConvert(const OpConverterParams& params) {
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/ops/unary_ops.cc:108]: s/ImplConvert/ConvertImpl/ to be consistent with how we use "Impl"?

**Review by bixia1 — COMMENTED** [review]
Looks good to me. Please also squash the commits into one.

**Comment by gbaned** [comment]
Hi @drivanov Can you please check @bixia1's comments and keep us posted ? Thank you!

File: tensorflow/compiler/tf2tensorrt/convert/ops/unary_ops.cc (hunk 2 above)
Comment (drivanov) [reply tensorflow/compiler/tf2tensorrt/convert/ops/unary_ops.cc:108]: Done

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc (hunk 1 above)
Comment (drivanov) [reply tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:1376]: Done

**Comment by drivanov** [comment]
@gbaned : I am done with the changes, suggested by @bixia1

**Comment by bixia1** [comment]
@drivanov thanks! would you please squash the commits into one?

**Comment by drivanov** [comment]
Replacing this one with [PR#55716](https://github.com/tensorflow/tensorflow/pull/55716)

**Pull request closed without merging**